scripts/validation_shell.js

```diff
@@ -0,0 +1,40 @@
+// Test a file is valid, by just loading it.
+
+// Shell integration.
+if (typeof console === 'undefined') {
+  console = { log: print };
+}
+var tempRet0;
+var binary;
+if (typeof process === 'object' && typeof require === 'function' /* node.js detection */) {
+  var args = process.argv.slice(2);
+  binary = require('fs').readFileSync(args[0]);
+  if (!binary.buffer) binary = new Uint8Array(binary);
+} else {
+  var args;
+  if (typeof scriptArgs != 'undefined') {
```

Comment:
What is `scriptArgs`? (I don't know much about the JS part, sorry)

Reply:
It's one of the ways various JS shells give us access to the commandline params. I think scriptArgs is for spidermonkey and arguments is for v8 (or maybe the opposite).